core/src/test/java/org/elasticsearch/search/sort/FieldSortBuilderTests.java

```diff
@@ -79,18 +84,23 @@ public FieldSortBuilder randomFieldSortBuilder() {
         if (randomBoolean()) {
             builder.sortMode(randomFrom(SortMode.values()));
         }
-
         if (randomBoolean()) {
             builder.setNestedSort(createRandomNestedSort(3));
         }
-
+        // the following are alternative ways to setNestedSort for nested sorting
+        if (randomBoolean()) {
+            builder.setNestedFilter(randomNestedFilter());
+        }
+        if (randomBoolean()) {
+            builder.setNestedPath(randomAlphaOfLengthBetween(1, 10));
```

Comment:
same here

Reply:
I added deprecation annotations to the setter and getter